Export variables for all Terraform Cloud workspaces and back up the files to Backblaze B2

dalenewby opened the pull request:
## Added

* Optionally suppress the `curl` progress bar output from `tfc-dump.pl`.
* Include example using a Docker image to export Terraform Cloud variables and back up the resulting files to a Restic repository on a Backblaze B2 bucket.

dalenewby committed:
Optionally suppress curl progress bar

dalenewby committed:
Store backups in a Restic repo on Backblaze B2

Dump variables for all workspaces and use Restic to back up the
resulting files to a repository stored in a Backblaze B2 bucket.

dalenewby committed:
Fix typos

dalenewby committed:
Fix typo

briskt commented on Dockerfile:
```diff
@@ -0,0 +1,29 @@
+
+# Variables set with ARG can be overridden at image build time with
+# "--build-arg var=value".  They are not available in the running container.
+ARG restic_ver=0.15.2
```
Probably not significant, but it caught my eye that lines 5 and 6 are different in that one uses quotes and the other doesn't

briskt commented on README.md:
```diff
@@ -37,3 +37,17 @@ Spaces in the variable set name are replaced with hyphens (`-`).
+1. Set the values for the variables contained in `local.env`.
+1. Obtain a Terraform Cloud access token. Go to https://app.terraform.io/app/settings/tokens to create an API token.
+1. Add the access token value to `local.env`.
+1. Obtain a Backblaze Application Key. Ensure it has these capabilities: deleteFiles, listBuckets, listFiles, readBuckets, readFiles, writeBuckets, writeFiles.
```
Thanks for specifying the needed permissions

dalenewby replied:
It is important when you create the Application Key with Terraform.  The Backblaze GUI only give you a choice between `Read`, `Write` and `Read and Write`.  The `Read and Write` choice gives you more capabilities than you need; I didn't see a way to reduce the capability list.

briskt commented on application/init.sh:
```diff
@@ -0,0 +1,19 @@
+echo "tfc-backup: init: Started"
+
+start=$(date +%s)
+/usr/local/bin/restic init || STATUS=$?
```
just a thought, and I'm not sure if it's even available in alpine, but it would be a lot simpler to use `time` to report the execution time, e.g. `time /usr/local/bin/restic init || STATUS=$?`. And come to think of it, do you really need to specify the /usr/local/bin path?

dalenewby replied:
`time` gives you four lines of output and I didn't want to parse the output.  I'm only including the the execution time in case a future user is trying to discover why the overall backup is taking so long.

I tend to include the path to executable programs when they aren't included in `/bin` or `/usr/bin`.  I've seen too many Unix-like systems that didn't include `/usr/local/bin` in the default `PATH` for all users.

briskt commented on tfc-backup.sh:
```diff
@@ -0,0 +1,75 @@
+# July 20, 2023
+
+# Required environment variables:
+# ATLAS_TOKEN        - Terraform Cloud access token
```
Just opinion, but I don't like to see duplicated information since humans tend to not keep duplicated information in sync very well. But you're welcome to leave this documentation in two places if you want.

dalenewby replied:
Humans can't keep program documentation in sync with the code, either, and I've submitted PRs to fix the ones I've discovered.  ;-)  I added the required variables to the code as I wrote it to remind me of what I needed to set to run the script.  The descriptions got added to `local.env.dist` to remind me of what values I needed to find as I prepared to run the script.  I thought about adding them to the `README.md`, too.

dalenewby committed:
Remove quotes around ARG values and apk cache files

dalenewby committed:
Doc update: create bucket before application key

Creating the Backblaze bucket first allows us to restrict the
application key to access only that bucket.
